backend/workflow_manager/workflow_v2/models/execution.py

```
@@ -313,6 +316,31 @@ def update_execution(
+            # Check if this is an API deployment execution
+            api_deployment = APIDeployment.objects.filter(id=self.pipeline_id).first()
+            if api_deployment and api_deployment.organization:
+                from api_v2.rate_limiter import APIDeploymentRateLimiter
+
+                APIDeploymentRateLimiter.release_slot(
+                    str(api_deployment.organization.organization_id), str(self.id)
+                )
+        except Exception as e:
+            # Log but don't fail the execution update for rate limit release errors
+            logger.exception(
+                f"Failed to release rate limit slot for execution {self.id}: {e}"
```

Ruff (0.14.3)

Redundant exception object included in `logging.exception` call

(TRY401)